core/pta/k3/otp.c

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Texas Instruments System Control Interface Driver
+ *
+ * Copyright (C) 2023 Texas Instruments Incorporated - https://www.ti.com/
+ *	Manorit Chawdhry <m-chawdhry@ti.com>
+ */
+
+#include <drivers/ti_sci.h>
+#include <inttypes.h>
+#include <k3/otp_keywriting_ta.h>
+#include <kernel/pseudo_ta.h>
+
+static TEE_Result write_otp_row(uint32_t param_types, TEE_Param params[4])
+{
+	int ret;
+	const uint32_t exp_param_types =
+		TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INPUT,
+				TEE_PARAM_TYPE_VALUE_INPUT, TEE_PARAM_TYPE_NONE,
+				TEE_PARAM_TYPE_NONE);
+
+	/*
+	 * Safely get the invocation parameters
+	 */
+	if (param_types != exp_param_types)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	ret = ti_sci_write_otp_row(params[0].value.a, params[1].value.a,
+				   params[1].value.b);
+	if (ret)
+		return ret;
+
+	IMSG("Written the value: 0x%08x", params[1].value.a);
```

Comment:
`	IMSG("Written the value: 0x%08"PRIx32, params[1].value.a);`
I think `DMSG()` would be better fit.

ditto in other added functions.